### DISCUSSION TITLE
Can I use scenic without rails?

### DISCUSSION THREAD

**Discussion by jtrupiano** [scenic-views/scenic#352] [Q&A] [answered]
I have an older app that I've stuffed a half dozen views into (using raw #exec's in old migrations).  I realize now that my schema.rb is missing these views.  I want to rectify that, and that's what has led me to discover scenic.

I thought that just adding "require 'scenic'" to my Rakefile and then running rake db:schema:dump would find the views that already exist in my database and include them in the newly generated schema.rb.  But that doesn't appear to be happening.  The new schema.rb is still the same as the old.  Do I need to include some sort of initialization hook in my Rakefile?

I am not using rails, just ActiveRecord and the standalone_migrations gem (plus foreigner too).  Am I going to run into issues with scenic because I am not also using rails?

Thanks in advance.

-John

**Comment by derekprior** [comment, marked as answer]
You will also need to call `Scenic.load` somewhere in order to load our ActiveRecord monkey patches (the only way something like Scenic can work, unfortunately). This is done automatically in Rails by virtue of the Railties code. You can see what that method does here: https://github.com/scenic-views/scenic/blob/be709057f2c7381b5d427f5492f8b9a6b39864d4/lib/scenic.rb#L16-L24

It's possible there are other dependencies on Rails that have inadvertently snuck in, so if you find any let us know here.

**Comment by jtrupiano** [comment]
That did the trick.  I included `Scenic.load` in my Rakefile and that monkeypatched the SchemaDumper as intended.  So far so good, not seeing any other rails dependencies.  But I'll drop a note back here if I run into something like that.

Thanks!
